src/EditorFeatures/VisualBasicTest/Diagnostics/RemoveAsyncModifier/RemoveAsyncModifierTests.vb

```diff
@@ -0,0 +1,306 @@
+            Await VerifyVB.VerifyCodeFixAsync(
+"Imports System
+Imports System.Threading.Tasks
+
+Class C
+    Sub Goo()
+        dim f as Func(of Task(Of Integer)) =
+            Async {|BC42356:Function|}()
+                Return 1
+            End Function
+    End Sub
+End Class",
+"Imports System
+Imports System.Threading.Tasks
+
+Class C
+    Sub Goo()
+        dim f as Func(of Task(Of Integer)) =
+            Function()
+                Return Task.FromResult(1)
+            End Function
+    End Sub
+End Class")
+        End Function
+
+        <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveAsyncModifier)>
+        Public Async Function Sub_Missing() As Task
+            Dim source = "
+Imports System
+
+Class C
+    Async Sub {|BC42356:Goo|}()
+        Console.WriteLine(1)
+    End Sub
+End Class"
+            Dim expected = "
+Imports System
+
+Class C
+    Async Sub {|#0:Goo|}()
+        Console.WriteLine(1)
+    End Sub
+End Class"
+
+            Dim test = New VerifyVB.Test()
+            test.TestState.Sources.Add(source)
+            test.FixedState.Sources.Add(expected)
+            ' /0/Test0.vb(5) : warning BC42356: This async method lacks 'Await' operators and so will run synchronously. Consider using the 'Await' operator to await non-blocking API calls, or 'Await Task.Run(...)' to do CPU-bound work on a background thread.
+            test.FixedState.ExpectedDiagnostics.Add(DiagnosticResult.CompilerWarning("BC42356").WithSpan(5, 15, 5, 18))
+            Await test.RunAsync()
+        End Function
+
+        <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveAsyncModifier)>
+        Public Async Function MultiLineSubLambda_Task_Missing() As Task
+            Dim source = "
+Imports System
+
+Class C
+    Sub Goo()
+        dim f as Action =
+            Async {|BC42356:Sub|}()
+                Console.WriteLine(1)
+            End Sub
+    End Sub
+End Class"
+            Dim expected = "
+Imports System
+
+Class C
+    Sub Goo()
+        dim f as Action =
+            Async {|#0:Sub|}()
+                Console.WriteLine(1)
+            End Sub
+    End Sub
+End Class"
+
+            Dim test = New VerifyVB.Test()
+            test.TestState.Sources.Add(source)
+            test.FixedState.Sources.Add(expected)
+            ' /0/Test0.vb(7) : warning BC42356: This async method lacks 'Await' operators and so will run synchronously. Consider using the 'Await' operator to await non-blocking API calls, or 'Await Task.Run(...)' to do CPU-bound work on a background thread.
+            test.FixedState.ExpectedDiagnostics.Add(DiagnosticResult.CompilerWarning("BC42356").WithSpan(7, 19, 7, 22))
+            Await test.RunAsync()
+        End Function
+
+        <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveAsyncModifier)>
+        Public Async Function SingleLineSubLambda_Task_Missing() As Task
+            Dim source = "
+Imports System
+
+Class C
+    Sub Goo()
```

Comment:
I don't see any tests for preserving comment and Line Continuation
```
        dim f as Action = _
 _  ' This is a comment
            Async {|#0:Sub|}()

```

Reply:
note: i would consider these outlier cases. i.e. they're far outside of hte 99% case.  So preservation here would not be a priority.